src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -0,0 +1,495 @@
+    def acceptDrop(self, event):
+        if event.mimeData().hasFormat("application/x-silx-uri"):
+            byteString = event.mimeData().data("application/x-silx-uri")
+            url = silx.io.url.DataUrl(byteString.data().decode("utf-8"))
+            with silx.io.open(url.file_path()) as file:
+                data = file[url.data_path()]
+                if (
+                    silx.io.is_dataset(data)
+                    and data.ndim == 1
+                    and not self.model().fileExists(url.data_path())
+                ):
+                    event.acceptProposedAction()
+        else:
+            event.ignore()
+
+    def setX(self, url):
+        targetNode = "X"
+        node = self.model().getXParent()
+        self.model().addUrl(url, targetNode)
+        self._createRemoveButton(node, None)
+        self.model()._plot1D.resetZoom()
+
+    def addY(self, url):
+        targetNode = "Y"
+        node = self.model().getYParent()
+        self.model().addUrl(url, targetNode)
+        self._createIconWidget(node.rowCount() - 2, node)
+        self._createRemoveButton(node.rowCount() - 2, node)
+        self.model()._plot1D.resetZoom()
+
+    def clear(self):
+        self.model().clearAll()
+        index = self.model().index(0, 2)
+        self.setIndexWidget(index, None)
+
+    def setDropHighlight(self, value):
+        xDropFileItem = self.model().itemFromIndex(
+            self.model().sibling(0, 1, self.model().indexFromItem(self.model().getXParent()))
+        )
+        xDropFileItem.setData(value == "X", _DROP_HIGHLIGHT_ROLE)
+        yDropFileItem = self.model().getYParent().child(self.model().getYParent().rowCount() -1, 1)
+        yDropFileItem.setData(value == "Y", _DROP_HIGHLIGHT_ROLE)
+
+
+class DropPlot1D(plot.Plot1D):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setAcceptDrops(True)
+        self._treeView = None
+
+    def setTreeView(self, treeView):
+        self._treeView = treeView
+
+    def dragEnterEvent(self, event):
+        super().dragEnterEvent(event)
+        self._treeView.acceptDrop(event)
+
+    def dropEvent(self, event):
+        super().dropEvent(event)
+        byteString = event.mimeData().data("application/x-silx-uri")
+        url = silx.io.url.DataUrl(byteString.data().decode("utf-8"))
+
+        plotArea = self.getWidgetHandle()
+        dropPosition = plotArea.mapFrom(self, event.pos())
+
+        plotBounds = self.getPlotBoundsInPixels()
+        left, top, width, height = plotBounds
+        yAreaTop = top + height
+
+        if dropPosition.y() > yAreaTop:
+            targetNode = "X"
+        else:
+            targetNode = "Y"
+
+        with silx.io.open(url.file_path()) as file:
+            data = file[url.data_path()]
+            if silx.io.is_dataset(data) and data.ndim == 1:
+                if targetNode == "X":
+                    self._treeView.setX(url)
+                else:
+                    self._treeView.addY(url)
```

Comment:
Since the validity of the test was tested in `acceptDrop`, there is no need to test it here, so, unless I missed something, the code can be simplified:

```suggestion
        if dropPosition.y() > yAreaTop:
            self._treeView.setX(url)
        else:
            self._treeView.addY(url)
```